GHzDACs/JumpTableIntro.md

```diff
@@ -0,0 +1,152 @@
+
+There are now two pointers that can be manipulated:
+
+* The SRAM pointer determines what the DACs are outputting.
+  * This holds a memory address. The DACs play what is at this address, and the pointer gets incremented sequentially.
+* The jump table pointer holds the index of the currently active jump table operation.
+  * A jump table operation has a few parts:
+    * The operation code defines the type of command: e.g. IDLE, CHECK, CYCLE, JUMP, NOP, END
+    * The fromAddress is an SRAM address. When the SRAM pointer reaches the fromAddress, this jump table operation is called, and will execute on the _next_ cycle (i.e. the SRAM pointer will be at fromAddress+1 when the jump table operation executes).
+    * The toAddress is an SRAM address that is used in the CHECK, CYCLE, and JUMP operations.
+  * Jump table operations can manipulate both the SRAM pointer and the jump table pointer.
+
+# Jump Table Operations
+
+Let's make this concrete by enumerating the allowed jump table operations. Each jump table operation is given by a two byte operation code (opcode), as well as two three-byte SRAM addresses, the fromAddress and toAddress. The first byte of the opcode, Byte 0, gives the command and (possibly) arguments to it. The second byte, Byte 1, is commonly devoted to a jump table index. (NB: everything is little endian. "First byte" means least significant byte.) The encoding used below is copied from John's documentation; for example, `xxjjjjjj` means the first two bits `x` are ignored, while the six bits `j` form a single number, the use of which is given in the description. For details of actually defining the jump table, see the section "Jump Table Write" below.
+
+ Name |   Byte 1   |   Byte 0   | Description
+------|------------|------------|-------------
+IDLE  | `dddddddd` | `ddddddd0` | Wait for n+1 cycles, with n defined by the fifteen bits `d`. SRAM pointer will remain at fromAddress+1.
+NOP   | `xxxxxxxx` | `xxxx0101` | Null operation: move both the SRAM pointer and the jump table pointer forward by one.
+CHECK | `xxjjjjjj` | `iiiin001` | Query the daisy chain bit at index `iiii`. If it is equal to `n`, move the SRAM pointer to the toAddress and the jump table pointer to the jump table index indicated by `jjjjjj`. If not, move both the SRAM pointer and the jump table pointer forward by one.
+JUMP  | `xxjjjjjj` | `xxxx1101` | Move the SRAM pointer to the toAddress, and the jump table pointer to `jjjjjj`.
+CYCLE | `xxjjjjjj` | `xxccx011` | Compare the counter* (index given by `cc`) with its countTo parameter (also index `cc`). If false (counter != countTo): increment counter, move SRAM pointer to toAddress, move jump table pointer to `jjjjjj`. If true (counter == countTo): reset counter to 0, move both SRAM pointer and jump table pointer forward by one.
+END   | `xxxxxxxx` | `xxxxx111` | End. The SRAM pointer will stop and idle at fromAddress + 2.
```

Comment:
can we clarify here, do `x`s actually mean those bits are ignored? We set them to zero by convention, but is that required?